src/ceph-volume/ceph_volume/util/disk.py

```diff
@@ -747,16 +747,11 @@ def get_devices(_sys_block_path='/sys/block'):
     for block in block_devs:
         devname = os.path.basename(block[0])
         diskname = block[1]
-        if block[2] not in ['disk', 'mpath']:
+        if block[2] == 'part' or block[2] == 'loop':
```

Comment:
```suggestion
        if block[2] not in ['disk', 'mpath', 'crypt']:
```
I think we should play it save here and explicitly allow types.

Reply:
I want to add "lvm" too, it OK? It's essential for container-based Ceph orchestration, like Rook and cephadm, to prevent modifying LVM metadata by host and Ceph container simultaneously. Actually Rook has already hit this problem.

https://github.com/rook/rook/blob/master/Documentation/ceph-common-issues.md#lvm-metadata-can-be-corrupted-with-osd-on-lv-backed-pvc

```
There is a critical flaw in OSD on LV-backed PVC. LVM metadata can be corrupted if both the host and OSD container modify it simultaneously. For example, the administrator might modify it on the host, while the OSD initialization process in a container could modify it too. In addition, if lvmetad is running, the possibility of occurence gets higher. In this case, the change of LVM metadata in OSD container is not reflected to LVM metadata cache in host for a while.
```

Reply:
OK. I'll change my PR as you suggested.

BTW, I want to contribute to resolving LVM related subjects. If possible, please let me know what are the known problems and anything I can do.